### ISSUE TITLE
Kata not able to handle systemd cgroups correctly

### ISSUE THREAD

**Issue by egernst** [kata-containers/runtime#2182]
# Description of problem

*This may not be an issue with Kata*

When validating SandboxCgroup Only with Kata, containerd 1.2.7 was utilized, and the cgroup was created in the path:

```
parent-cgroup-path=/kubepods.slice/kubepods-pod8949946d_3d17_415c_89d4_a8150388e128.slice/d22c91ffc5d7fa128db2a284baa50623935be294edda4d31a83b02838ef52cd3
```

In 1.3, this is: 
```
parent-cgroup-path="/vc/kubepods-podc9c28094_37b3_49e8_a9a7_939773aea4d2.slice:cri-containerd:56ff8b750887f33a7fd0dfe8110267bf957eb6f2d38ee43fd5802d34bd83a497"
```
This is observed when looking @ https://github.com/kata-containers/runtime/blob/master/virtcontainers/sandbox.go#L2180-L2208

With the following applied:
```go
--- a/virtcontainers/sandbox.go
+++ b/virtcontainers/sandbox.go
@@ -2192,6 +2192,11 @@ func (s *Sandbox) setupSandboxCgroup() error {
        }
        validContainerCgroup := utils.ValidCgroupPath(spec.Linux.CgroupsPath)
 
+       s.Logger().WithFields(logrus.Fields{
+               "parent-cgroup-path": validContainerCgroup,
+               "from-spec":          spec.Linux.CgroupsPath,
+       }).Warning("EGERNST: setupSandboxCgroup")
+
```

The spec.LinuxCgroupsPath looks correct, while the "ValudCgroupPath" provides an unexpected result (@ /vc/)

It looks like in 1.2.7 an absolute path was provided, whereas in 1.3.0 its a relative path (among other changes):

1.2.7:
```
/kubepods.slice/kubepods-podfoo.slice/bar
```
1.3.0:
```
kubepods-podfoo.slice:cri-containerd:bar
```

We need to either determine the path another way (in case of shim v2 are we called from appropriate path?) or work to get this provider as an absolute path again on containerd side. I'd prefer we work with an asbolute path so we can be assured we are within the correct (pod) cgroup created by Kubelet.

Per [runtime spec](https://github.com/opencontainers/runtime-spec/blob/master/config-linux.md#cgroups-path) relative path is acceptable, but when integrating with Kubernetes, this is problematic, as we want to make sure the pod is created within the cgroup created by Kubelet. 

In particular, from spec:
```
In the case of a relative path (not starting with /), the runtime MAY interpret the path relative to a runtime-determined location in the cgroups hierarchy.
```
In case of Kata, we don't have insight into what relative path would be appropriate, and as a result just default to /vc/*

# Expected result

we land appropriately into kubepod path...

# Actual result

we don't

**Comment by egernst** [comment]
Turns out that in 1.2.7, while Kubernetes was operating with systemd based cgroups, we had configured containerd to utilize cgroupfs.  As a result, a full path was provided.

After chatting with @random-liu, I now understand that its common practice to pass only the last component of the cgroup path to systemd.  See https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/dockershim/docker_service.go#L472
pkg/kubelet/dockershim/docker_service.go:472
